File: android/app/src/main/kotlin/pl/kwasow/sunshine/data/Wish.kt
package pl.kwasow.sunshine.data

import kotlinx.serialization.Serializable

@Serializable
data class Wish(
    val id: Int,
    val author: String,
    val content: String,
    val done: Boolean,
    val timestamp: Long,
) {
    // ====== Public methods
    fun update(
        newContent: String = content,
        newDone: Boolean = done,
    ): Wish = Wish(id, author, newContent, newDone, timestamp)
}

File: backend/src/entities/wish.php
<?php

class Wish implements JsonSerializable
{
    private $id;
    private $author;
    private $content;
    private $done;
    private $timestamp;

    public function __construct($id, $author, $content, $done, $timestamp)
    {
        $this->id = $id;
        $this->author = $author;
        $this->content = $content;
        $this->done = $done;
        $this->timestamp = $timestamp;
    }

    public function jsonSerialize()
    {
        return [
        'id' => $this->id,
        'author' => $this->author,
        'content' => $this->content,
        'done' => $this->done,
        'timestamp' => $this->timestamp
        ];
    }
}
